Makefile.pre.in

```diff
@@ -1061,69 +1064,69 @@ Programs/_freeze_module: Programs/_freeze_module.o $(LIBRARY_OBJS_OMIT_FROZEN)
-	Programs/_freeze_module site $(srcdir)/Lib/site.py Python/frozen_modules/site.h
+Python/frozen_modules/site.h: $(FREEZE_MODULE) Lib/site.py
+	$(FREEZE_MODULE) site $(srcdir)/Lib/site.py Python/frozen_modules/site.h
 
-Python/frozen_modules/stat.h: Programs/_freeze_module Lib/stat.py
-	Programs/_freeze_module stat $(srcdir)/Lib/stat.py Python/frozen_modules/stat.h
+Python/frozen_modules/stat.h: $(FREEZE_MODULE) Lib/stat.py
+	$(FREEZE_MODULE) stat $(srcdir)/Lib/stat.py Python/frozen_modules/stat.h
 
-Python/frozen_modules/__hello__.h: Programs/_freeze_module Lib/__hello__.py
-	Programs/_freeze_module __hello__ $(srcdir)/Lib/__hello__.py Python/frozen_modules/__hello__.h
+Python/frozen_modules/__hello__.h: $(FREEZE_MODULE) Lib/__hello__.py
+	$(FREEZE_MODULE) __hello__ $(srcdir)/Lib/__hello__.py Python/frozen_modules/__hello__.h
 
-Python/frozen_modules/__phello__.h: Programs/_freeze_module Lib/__phello__/__init__.py
-	Programs/_freeze_module __phello__ $(srcdir)/Lib/__phello__/__init__.py Python/frozen_modules/__phello__.h
+Python/frozen_modules/__phello__.h: $(FREEZE_MODULE) Lib/__phello__/__init__.py
+	$(FREEZE_MODULE) __phello__ $(srcdir)/Lib/__phello__/__init__.py Python/frozen_modules/__phello__.h
 
-Python/frozen_modules/__phello__.ham.h: Programs/_freeze_module Lib/__phello__/ham/__init__.py
-	Programs/_freeze_module __phello__.ham $(srcdir)/Lib/__phello__/ham/__init__.py Python/frozen_modules/__phello__.ham.h
+Python/frozen_modules/__phello__.ham.h: $(FREEZE_MODULE) Lib/__phello__/ham/__init__.py
+	$(FREEZE_MODULE) __phello__.ham $(srcdir)/Lib/__phello__/ham/__init__.py Python/frozen_modules/__phello__.ham.h
 
-Python/frozen_modules/__phello__.ham.eggs.h: Programs/_freeze_module Lib/__phello__/ham/eggs.py
-	Programs/_freeze_module __phello__.ham.eggs $(srcdir)/Lib/__phello__/ham/eggs.py Python/frozen_modules/__phello__.ham.eggs.h
+Python/frozen_modules/__phello__.ham.eggs.h: $(FREEZE_MODULE) Lib/__phello__/ham/eggs.py
+	$(FREEZE_MODULE) __phello__.ham.eggs $(srcdir)/Lib/__phello__/ham/eggs.py Python/frozen_modules/__phello__.ham.eggs.h
 
-Python/frozen_modules/__phello__.spam.h: Programs/_freeze_module Lib/__phello__/spam.py
-	Programs/_freeze_module __phello__.spam $(srcdir)/Lib/__phello__/spam.py Python/frozen_modules/__phello__.spam.h
+Python/frozen_modules/__phello__.spam.h: $(FREEZE_MODULE) Lib/__phello__/spam.py
+	$(FREEZE_MODULE) __phello__.spam $(srcdir)/Lib/__phello__/spam.py Python/frozen_modules/__phello__.spam.h
 
-Python/frozen_modules/frozen_only.h: Programs/_freeze_module Tools/freeze/flag.py
-	Programs/_freeze_module frozen_only $(srcdir)/Tools/freeze/flag.py Python/frozen_modules/frozen_only.h
+Python/frozen_modules/frozen_only.h: $(FREEZE_MODULE) Tools/freeze/flag.py
+	$(FREEZE_MODULE) frozen_only $(srcdir)/Tools/freeze/flag.py Python/frozen_modules/frozen_only.h
 
 # END: freezing modules
 
-Tools/scripts/freeze_modules.py: Programs/_freeze_module
+Tools/scripts/freeze_modules.py: $(FREEZE_MODULE)
```

Comment:
I guess when Make sees a reference to something outside the tree it's just going to take it at face value, assuming it exists at all? What's the failure mode if that file doesn't exist?

Reply:
Our Makefile does not specify a build rule for the target ``$(FREEZE_MODULE)``. There is only a build rule for ``Programs/_freeze_module``. If ``FREEZE_MODULE`` points to a different, non-existing file then make would complain and stop:

```
$ make regen-deepfreeze  FREEZE_MODULE=builddep/origin/Programs/invalid
make: *** No rule to make target 'builddep/origin/Programs/invalid', needed by 'Python/frozen_modules/importlib._bootstrap.h'.  Stop.
```